tools/wptrunner/wptrunner/wptrunner.py

```diff
@@ -235,143 +360,83 @@ def run_tests(config, test_paths, product, **kwargs):
-                        except KeyboardInterrupt:
-                            logger.critical("Main thread got signal")
-                            manager_group.stop()
-                            raise
-                        test_count += manager_group.test_count()
-                        unexpected_count += manager_group.unexpected_count()
-                        unexpected_pass_count += manager_group.unexpected_pass_count()
                 recording.set(["after-end"])
-                test_total += test_count
-                unexpected_total += unexpected_count
-                unexpected_pass_total += unexpected_pass_count
-                logger.info("Got %i unexpected results, with %i unexpected passes" %
-                            (unexpected_count, unexpected_pass_count))
+                logger.info(
+                    "Got %i unexpected results, with %i unexpected passes" %
+                    (counts["unexpected"], counts["unexpected_pass"]))
                 logger.suite_end()
-                if repeat_until_unexpected and unexpected_total > 0:
-                    break
-                if repeat_count == 1 and len(test_loader.test_ids) == skipped_tests:
-                    break
 
-    if test_total == 0:
-        if skipped_tests > 0:
-            logger.warning("All requested tests were skipped")
-        else:
-            if kwargs["default_exclude"]:
-                logger.info("No tests ran")
-                return True
-            else:
-                logger.critical("No tests ran")
-                return False
+                # Note this iteration's runtime and pad the total time
+                # by 10% to ensure ample time for the next iteration(s).
+                iteration_runtime = datetime.now() - iteration_start
+                iteration_runtime = timedelta(
+                    seconds=(iteration_runtime.total_seconds() * 1.1))
 
-    if unexpected_total and not kwargs["fail_on_unexpected"]:
-        logger.info("Tolerating %s unexpected results" % unexpected_total)
-        return True
+                # determine the longest test suite runtime seen.
+                longest_iteration_time = max(longest_iteration_time,
+                                             iteration_runtime)
 
-    all_unexpected_passed = (unexpected_total and
-                             unexpected_total == unexpected_pass_total)
-    if all_unexpected_passed and not kwargs["fail_on_unexpected_pass"]:
-        logger.info("Tolerating %i unexpected results because they all PASS" %
-                    unexpected_pass_total)
-        return True
+                if repeat_until_unexpected and counts["unexpected"] > 0:
+                    break
+                if counts["repeat"] == 1 \
+                        and len(test_loader.test_ids) == counts["skipped"]:
+                    break
 
-    return unexpected_total == 0
+    return evaluate_runs(counts, iteration_timeout, kwargs)
```

Comment:
I think we should make `run_tests` return `counts` directly. Then `start` can discard the extra information and we can avoid putting the output in a kwarg.

Reply:
Yes, looking at this possible solution, it's a relatively simple fix that can be easily implemented and will not affect other tests using wptrunner. I made a change to only return `counts["repeat"]` to only give back the number of iterations of the test suite that were run, rather than all of `counts`. Do you think this is a good idea? Or would it be functionally better to return all of `counts` in the chance that other uses might arise and need further access to the other counts?